src/frontend/src/components/SearchBar/SearchBar.jsx

```diff
@@ -39,16 +39,12 @@ const useStyles = makeStyles((theme) => ({
 
 function CustomizedInputBase(props) {
   const classes = useStyles();
-  const [state, setState] = React.useState({
-    filter: '',
-  });
-  const { searchText, onChangeHandler } = props;
+  const [selectedFilter, setSelectedFilter] = useState(props.filter);
```

Comment:
This is an anit-pattern: you don't mix props and state.  If we're receiving this on props from our parent, we just use that as is, without putting into state here.